### PR TITLE
Feature/hijack header

### PR DESCRIPTION
**Pull request by annashamray**
Issue - https://github.com/GemeenteUtrecht/ZGW/issues/835 (BE)

### CODE REVIEW

File: backend/src/zac/accounts/middleware.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+
+        # add hijack header for FE
+        is_hijacked = request.session.get("is_hijacked_user", False)
+        response[self.header] = is_hijacked
```
Comment (sergei-maertens) [inline backend/src/zac/accounts/middleware.py:15]:
```suggestion
        response[self.header] = "true" if is_hijacked else "false"
```

Let's not rely on implicit type conversions, this is also consistent with https://github.com/GemeenteUtrecht/zaakafhandelcomponent/blob/main/backend/src/zac/contrib/kownsl/views.py#L110

File: backend/src/zac/api/schema.py [hunk 2]
```diff
@@ -35,3 +39,16 @@ def get_summary(self):
...
         return action_or_method_summary or cls_summary
+
+    def get_override_parameters(self):
+        """ Add hijack header"""
+        hijack_response_header = OpenApiParameter(
+            name=settings.HIJACK_HEADER,
+            type=str,
+            location=OpenApiParameter.HEADER,
+            description=_("Header displays if the user is hijacked."),
+            enum=["False", "True"],
```
Comment (sergei-maertens) [inline backend/src/zac/api/schema.py:50]:
```suggestion
            enum=["false", "true"],
```

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: backend/src/zac/accounts/tests/test_hijack_header.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+        for url in self.urls:
+            with self.subTest(url):
+                response = self.client.get(url)
+
+                self.assertEqual(
+                    response.status_code, status.HTTP_200_OK, response.json()
+                )
+                self.assertIn(settings.HIJACK_HEADER, response)
+                self.assertEqual(response[settings.HIJACK_HEADER], "True")
```
Comment (damm89) [inline backend/src/zac/accounts/tests/test_hijack_header.py:54]: After @sergei-maertens' suggestions: 
```suggestion
                self.assertEqual(response[settings.HIJACK_HEADER], "true")
```

File: backend/src/zac/accounts/tests/test_hijack_header.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+    def test_response_not_hijacked(self):
+        for url in self.urls:
+            with self.subTest(url):
+
+                response = self.client.get(url)
+
+                self.assertEqual(response.status_code, status.HTTP_200_OK)
+                self.assertIn(settings.HIJACK_HEADER, response)
+                self.assertEqual(response[settings.HIJACK_HEADER], "False")
```
Comment (damm89) [inline backend/src/zac/accounts/tests/test_hijack_header.py:38]: After @sergei-maertens' suggestions: 
```suggestion
                self.assertEqual(response[settings.HIJACK_HEADER], "false")
```

**Review by damm89 — APPROVED** [review]

**Review by sergei-maertens — CHANGES_REQUESTED** [review]
~~I don't see any changes to the `zac.conf.base` module to add the `HIJACK_HEADER` setting and it doesn't look like a default setting from Hijack either. Can you add it?~~

I need more coffee before I review code

**Review by sergei-maertens — APPROVED** [review]

**Pull request merged**